src/router/storeSync/storeSync.routerPlugin.js

```diff
@@ -232,22 +251,31 @@ const storeSyncRouterPlugin = (router, store) => {
     // which was LEGACY. By moving this subscription to the after Each loop, we ensure the 'currentRoute'
     // is always set to MAPVIEW, avoiding a lock of the viewer.
     router.afterEach((to) => {
+        const logPrefix = '[Router store plugin/afterEach]'
         if (MAP_VIEWS.includes(to.name) && !unsubscribeStoreMutation) {
-            log.info('[Router store plugin] Entering MapView, register store mutation watcher')
+            log.info(`${logPrefix}MapView entered, register store mutation watcher`)
             // listening to store mutation in order to update URL
             unsubscribeStoreMutation = store.subscribe((mutation) => {
                 if (mutation.type === 'setAppIsReady') {
-                    log.info(
-                        '[Router store plugin] app is ready, trigger initial URL query watcher'
-                    )
-                    const newRoute = urlQueryWatcher(store, to)
-                    if (newRoute) {
-                        router.push(newRoute)
-                    }
+                    // If the app was not yet ready after entering the map view, we need to
+                    // trigger the initial urlQuery watcher otherwise we have a blank application.
+                    log.info(`[store sync router] App is ready, trigger initial URL query watcher`)
+                    initialUrlQueryWatcher(to, store, router)
```

Comment:
Most of the time the initial URL query watcher happen here, if you comment this line you have exactly the same behavior observed in ticket PB-747